portal-web/docroot/html/taglib/ui/input_date/page.jsp

```diff
@@ -46,7 +46,18 @@ String monthParamId = namespace + HtmlUtil.getAUICompatibleId(monthParam);
 String nameId = namespace + HtmlUtil.getAUICompatibleId(name);
 String yearParamId = namespace + HtmlUtil.getAUICompatibleId(yearParam);
 
-Calendar calendar = CalendarFactoryUtil.getCalendar(yearValue, monthValue, dayValue);
+Calendar calendar = null;
+
+if (required && (yearValue == 0) && (monthValue == -1) && (dayValue == 0)) {
```

Comment:
Wouldn't this be safer if it was a grouped OR operation? :thinking: 

```java
if ( required && ( (yearValue == 0) || (monthValue == -1) || (dayValue == 0) ) ) {
```

Reply:
Or doing it separately per variable...

Reply:
Would work also. I tried to generate just a minimal fix and only those three exact values means the overall default value, anything else would mean that at least one of the values has been set.

Reply:
I see... 

And in the case where one of the values is set but the others don't, will it work? 

I'm thinking about entering the [else branch of the if](https://github.com/liferay-frontend/liferay-portal/pull/745/files#diff-38dc9ea938856268a239c2ded3d9346c2a7680c6edcd954a0dfd549e48b6e9a7R59). That line is going to construct a `Calendar` instance for an invalid value and I'm not sure what side effects will that have (I mean, I don't know if it is safe to have a `Calendar` with `year == 0` or `month == -1`, ...).

If it is possible that the code enters that branch and that leads to an incorrect `Calendar` value I suppose we should also cover that scenario. Otherwise this will hit us again, I'm afraid...

Reply:
@izaera 

>And in the case where one of the values is set but the others don't, will it work?

It will generate a date for sure, as when year=0, month=-1, day=0 were used, probably not a meaningful one, but a valid date.

We could validate the arguments, although that will imply checking not only lower bounds, but also upper bounds (at least with month and day), but there's a slight difference with the current case: now they aren't using any of our input parameters, so is somehow our fault setting a date that requires multiple clicks to bring it to "now" that will be the most likely use. But if they start using the parameters assigning no reasonable values to day or month (or even year) we could argue that they aren't using them properly and in the end we're offering the same behavior as _java.util.GregorianCalendar_ when "weird" parameters are introduced.

Reply:
Sound very reasonable. Thanks for the explanation :-).

Reply:
Let's forward this. And sorry for the delay (I forgot to check your answers yesterday :-( )

Reply:
No worries! Thanks a lot for @izaera